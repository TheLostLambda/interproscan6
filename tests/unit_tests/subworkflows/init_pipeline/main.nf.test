nextflow_workflow {

    name "Test suborkflow INIT_PIPELINE"
    script "subworkflows/init/main.nf"
    workflow "INIT_PIPELINE"

    stage {
        symlink "tests/data/test.faa"
        symlink "conf/applications.config"
    }

    test("Should run without failures") {
        when {
            params {
                apps_config = [
                    pfam: [
                        name: "Pfam",
                        dir: "pfam",
                        hmm: "pfam_a.hmm",
                        seed: "pfam_a.seed",
                        clan: "pfam_clans",
                        dat: "pfam_a.dat",
                        has_data: true
                    ],
                    ncbifam: [
                        name: "NCBIFAM",
                        dir: "ncbifam",
                        hmm: "ncbifam.hmm",
                        has_data: true
                    ]
                ]
            }
            workflow {
                """
                input[0] = "meta/tests/data/test.faa"        // input
                input[1] = "pfam,ncbifam"                    // applictaions
                input[2] = params.apps_config                // apps_config
                input[3] = false                             // download
                input[4] = false                             // offline
                input[5] = "data"                            // datadir
                input[6] = "json,tsv,xml"                    // formats
                input[7] = "outdir"                          // outdir
                input[8] = "latest"                          // interpro_version
                input[9] = false                             // skip_interpro
                input[10] = false                            // goterms
                input[11] = false                            // pathways
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

}
