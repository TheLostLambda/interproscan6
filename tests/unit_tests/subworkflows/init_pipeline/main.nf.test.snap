{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "test.faa:md5,ca69be62364cac44c8083faace76014e"
                ],
                "1": [
                    [
                        "ncbifam",
                        "pfam"
                    ]
                ],
                "2": [
                    [
                        
                    ]
                ],
                "3": [
                    [
                        
                    ]
                ],
                "4": [
                    [
                        "TSV",
                        "XML",
                        "JSON"
                    ]
                ],
                "5": [
                    "latest"
                ],
                "apps": [
                    [
                        "ncbifam",
                        "pfam"
                    ]
                ],
                "datadir": [
                    [
                        
                    ]
                ],
                "fasta": [
                    "test.faa:md5,ca69be62364cac44c8083faace76014e"
                ],
                "formats": [
                    [
                        "TSV",
                        "XML",
                        "JSON"
                    ]
                ],
                "outdir": [
                    [
                        
                    ]
                ],
                "version": [
                    "latest"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-05-22T11:27:01.282968"
    }
}